### PR TITLE
NotificationsManager: Revamp Mark II

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Description:
In this PR we're moving over `Interactive Notifications` specific code over from *NotificationsManager* to *InteractiveNotificationsHandler* helper.

Plus: I've updated the documentation style of the new Notifications-Y handling classes.

Reference: #4069

#### Scenario 1: Comment Reply
1. Fresh install the app, and log into your WordPress.com account
2. Kill (or) close the app
3. Receive a comment in a blog that has comments auto-approval enabled
4. In the springboard, pull the notification to reveal its interactive actions
5. Tap over the *Reply* button

As a result, the app should launch, and the associated Notification should be onscreen (allowing the user to reply right there).

#### Scenario 2: Comment Like
1. Fresh install the app, and log into your WordPress.com account
2. Kill (or) close the app
3. Receive a comment in a blog that has comments auto-approval enabled
4. In the springboard, pull the notification to reveal its interactive actions
5. Tap over the *Like* button

As a result, the comment should receive a like.

#### Scenario 3: Comment Approval
1. Fresh install the app, and log into your WordPress.com account
2. Kill (or) close the app
3. Receive a comment in a blog that has comments auto-approval *disabled*
4. In the springboard, pull the notification to reveal its interactive actions
5. Tap over the *Approve* button

As a result, the comment should get approved.

Needs Review: @aerych 

Eric, may i bother you with a review?
Thanks in advance sir!

### CODE REVIEW

File: WordPress/Classes/Utility/PushNotificationsManager.swift [hunk 1]
```diff
@@ -59,41 +52,30 @@ final public class PushNotificationsManager : NSObject
...
-     *  @details    Both Helpshift and Mixpanel will also be initialized. The token will be persisted across
-     *              App Sessions.
-     */
+    /// Registers the Device Token agains WordPress.com backend, if there's a default account.
+    ///
+    /// - Note: Both Helpshift and Mixpanel will also be initialized. The token will be persisted across App Sessions.
+    ///
     func registerDeviceToken(tokenData: NSData) {
         // We want to register Helpshift regardless so that way if a user isn't logged in
         // they can still get push notifications that we replied to their support ticket.
```
Comment (aerych) [inline WordPress/Classes/Utility/PushNotificationsManager.swift]: This comment is for the two lines of code just below this one.
```
Helpshift.sharedInstance().registerDeviceToken(tokenData)
Mixpanel.sharedInstance().people?.addPushDeviceToken(tokenData)
```
We chatted briefly about this in Slack, just documenting here for posterity.  In both cases, `sharedInstance()` can return `nil` if the app was built without the necessary keys in .wpcom_app_credentials.  The result is the app will probably crash for most contributors with an  “unexpectedly found nil when unwrapping an optional” error.  

I know its not strictly a part of this PR, but maybe we could add some nil checks?  A separate PR would also be fine. :)

**Comment by aerych** [comment]
Hiya @jleandroperez !  Code looks good to me and I was able to test the above scenarios courtesy of a kindly assist by Mr. @roundhill.  I tested both from a notification on the lock screen, and from a notification at the top of my home screen.  

I chose the reply action first, which left me with the expectation that choosing Like would also open the app, so I was confused at first when that didn't happen.  Then I realized that the like happened anyway and the app wasn't supposed to launch.  Take that for whatever its worth. ;) 

I left a note about the crash case I found but since its not strictly a part of this PR we could handle that separately. Fine to :shipit: with a patch, or a patch in another PR.  Cheers!

File: WordPress/Classes/Utility/PushNotificationsManager.swift (hunk 1 above)
Comment (jleandroperez) [inline WordPress/Classes/Utility/PushNotificationsManager.swift]: **THANKS** for catching that Eric!!!!

**Comment by jleandroperez** [comment]
@aerych again, many many many thanks for the review (+ for catching that glitch!!)

**Pull request merged**